pkg/build/build.go

```diff
@@ -173,3 +173,29 @@ func CreateTarball(source, target string) error {
 			return err
 		})
 }
+
+func BuildS2I(image, context, builderImage string) error {
+	var cmd []string
+
+	// Generate the buildconfig and imagestream
+	log.Info("Creating buildconfig and imagestream")
+	if builderImage == "" {
+		cmd = []string{"oc", "new-build", "--binary", "--name=" + image}
```

Comment:
This command will create BuildConfig with DockerStrategy so no S2I :-(

Reply:
It might be good to do a small research first.

If you do `oc new-build --binary` without builder image (`--docker-image`) that it will always create BC with DockerStrategy, at least this is what I saw.

Uou can provide a directory to `oc new-build` and  it might crate BC with SourceStrategy (s2i) in some cases, or even jenkinsPipelineStrategy if it detects Jenkinsfile in the directory. If directory is in git that it will add git remote as source :-(  So far I couldn't figure out the way to force binary build with SourceStrategy using `oc new-build`

Reply:
there is a `--strategy` option that allows you to force SourceStragy, but than you are required to provide builder image

Reply:
If we can't find a way to do builder image autodetection with binary builds than I think that we can require manually selecting builder image. It will be the first step for S2I and than we can figure out autodetection later

Reply:
@kadel we can do one thing, just start the normal build and immediately trigger it using the source in local directory:

```conosle
oc new-build . --name=pythonapp --strategy=source
oc start-build pythonapp --from-dir=.
```
So we are achieving similar results as the binary build, we are triggering the build using the code provided locally.

Reply:
Not sure that this is a good idea. BuildConfig will be created with git as a source. 
If someone triggers BuildConfig build without source-dir it will build from git repository :-( and that might not be what they intended. Also when I tried that the first build from git was still triggered and finished, so we would have to cancel it before it finished. Only the build form local dir should finish successfully.

I'm not sure if there is a good solution if we want autodetection of builder image :-(
Have you checked how that code looks like in `oc`? Maybe we could just use autodetection part from it, and create BC ourselves.